report/5.Conclusion/FutureWork.tex

```diff
@@ -1,2 +1,35 @@
 % !TeX root = ../main.tex
 \section{Future work}
+
+\subsection{User-tests}
+As per the conclusion of this report, the HCL language has not been usability-tested.
+The target-group of HCL consists of two sub-groups, novices with no programming experience, and novices with limited programming experience.
+
+To conduct the usability-test, the sub-group with no programming experience, would have to be taught simple programming constructs in HCL and general programming paradigms.
+The sub-group with limited programming experience, would be asked to write simple programs in a language they know, and then write the equivalent programs in HCL.
+
+The first sub-group would be used to determine whether HCL is intuitive.
+The second sub-group would be used to evaluate whether HCL simplifies the programming paradigms enough.
+
+\subsection{Improvements to the language}
+While HCL is a complete language, there are still improvements that could be made.
+
+\textbf{Working Arduino}\\
+While the project group wanted HCL to run on the Arduino platform, this feature was not completed in time for the project hand-in.
+The Arduino does not allow lambdas, or the import of the library that is used in the C++ portion of the code.
```

Comment:
not allow for the std::function type from the "functional "C++ standard library.
(Lambdas are (as far as I have understood) supported, however not the data type we use.)
Apart from lambdas, the only library that we are missing is for smart pointers. However, the smart pointer we use could very easily be ported. The main issue is the lack of the std::function type.